lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3039,8 +3039,13 @@ ParseResult FIRStmtParser::parseMem(unsigned memIndent) {
                             ports, moduleContext.targetsInModule);
 
     auto sym = getSymbolIfRequired(annotations.first, id);
+    // Port annotations are an ArrayAttr of ArrayAttrs, so iterate over all the
+    // annotations for each port, and check if any of the port needs a symbol.
     if (!sym)
-      sym = getSymbolIfRequired(annotations.second, id);
+      for (auto portAnno : annotations.second.getAsRange<ArrayAttr>())
+        if ((sym = getSymbolIfRequired(portAnno, id)))
+          break;
```

Comment:
```suggestion
      for (auto portAnno : annotations.second.getAsRange<ArrayAttr>())
        if (sym = getSymbolIfRequired(portAnno, id))
          break;
```

Reply:
Causes a warning, `warning: using the result of an assignment as a condition without parentheses`. 
This was my first implementation, but modified it for the warning.

Reply:
This seems too weird, I think you should just split it into two lines